duffy/api_v2/views.py

```diff
@@ -0,0 +1,206 @@
+# -*- coding: utf-8 -*-
+import datetime
+
+from flask import Blueprint, request, jsonify, abort, current_app
+from functools import wraps
+from duffy.models import Openstack_Host, HostSchema, Session, SessionSchema, Project
+from duffy.database import db
+from credentials import get_nova_credentials
+from novaclient.Client import client
+
+blueprint = Blueprint('api_v1', __name__)
+
+
+def duffy_key_required(fn):
+    @wraps(fn)
+    def decorated(*args, **kwargs):
+        duffy_key = request.args.get('key', None)
+        if not duffy_key:
+            return jsonify({'msg': 'Invalid duffy key'}), 403
+        return fn(*args, **kwargs)
+    return decorated
+
+def ssid_required(fn):
+    @wraps(fn)
+    def decorated(*args, **kwargs):
+        ssid = request.args.get('ssid')
+        if not ssid:
+            return jsonify({'msg': 'Invalid session ID'}), 403
+
+        sess = Session.query.get(ssid)
+        if not sess:
+            return jsonify({'msg': 'Invalid session ID'}), 403
+        return fn(*args, **kwargs)
+    return decorated
+
+
+@blueprint.route('/Node/get')
+@duffy_key_required
+def nodeget():
+    get_image = int(request.args.get('image'))
+    get_key = request.args.get('key')
+    get_flavor = request.args.get('flavor')
+
+    project = Project.query.get(get_key)
+
+    if not project:
+        return 'Invalid duffy key'
+
+    if not get_image:
+        get_image = 'centos_7_x86_64'
+
+    if not get_flavor:
+        get_flavor = 'tiny'
+
+    five_minutes_ago = datetime.datetime.utcnow() - datetime.timedelta(minutes=5)
+    recent_sessions_count = Session.query.filter(Session.apikey==get_key,
+            Session.delivered_at > five_minutes_ago, (Session.state == 'Prod') \
+            | (Session.state == 'Fail')).count()
+    current_app.logger.info("Activity Rate: {} - {}".format(get_key,
+        recent_sessions_count))
+
+    if recent_sessions_count > 5:
+        return jsonify('Failed to allocate nodes: Rate limit exceeded: {} \
+                sessions in the last 5 minutes'.format(recent_sessions_count)\
+                       ), 429
+
+    try:
+        credentials = get_nova_credentials()
+        nova_client = Client(**credentials)
+
+        image = nova_client.glance.find_image(get_image)
+        flavor = nova_client.flavors.find(get_flavor)
+        name = image+str(int(time.time()))
+        instance = nova_client.servers.create(name=name, image=image,
+                flavor=flavor)
+        while(nova_client.servers.get(instance.id).status != 'ACTIVE'):
+            sys.stdout.write('waiting for vm to get into ACTIVE state')
```

Comment:
Let's use `current_app.logger` (see [views.py#L55](https://github.com/CentOS/duffy/blob/afc0683883fd8fa1bd22de1265f6d32fb74a5569/duffy/) for an example) in the DEBUG context instead of writing directly to stdout. Move this above your while loop, and add a logger statement after saying that we found the vm in the active state